uptime.go

```diff
@@ -0,0 +1,347 @@
+package godo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"path"
+)
+
+const uptimeChecksBasePath = "/v2/uptime/checks"
+
+// UptimeChecksService is an interface for creating and managing Uptime checks with the DigitalOcean API.
+// See: https://docs.digitalocean.com/reference/api/api-reference/#tag/Uptime
+type UptimeChecksService interface {
+	List(context.Context, *ListOptions) ([]UptimeCheck, *Response, error)
+	Get(context.Context, string) (*UptimeCheck, *Response, error)
+	GetUptimeCheckState(context.Context, string) (*UptimeCheckState, *Response, error)
+	Create(context.Context, *CreateUptimeCheckRequest) (*UptimeCheck, *Response, error)
+	Update(context.Context, string, *UpdateUptimeCheckRequest) (*UptimeCheck, *Response, error)
+	Delete(context.Context, string) (*Response, error)
+	GetUptimeAlert(context.Context, string, string) (*UptimeAlert, *Response, error)
+	ListUptimeAlerts(context.Context, string, *ListOptions) ([]UptimeAlert, *Response, error)
+	CreateUptimeAlert(context.Context, string, *CreateUptimeAlertRequest) (*UptimeAlert, *Response, error)
+	UpdateUptimeAlert(context.Context, string, string, *UpdateUptimeAlertRequest) (*UptimeAlert, *Response, error)
+	DeleteUptimeAlert(context.Context, string, string) (*Response, error)
+}
+
+// UptimeChecksServiceOp handles communication with Uptime Check methods of the DigitalOcean API.
+type UptimeChecksServiceOp struct {
+	client *Client
+}
+
+// UptimeCheck represents a DigitalOcean UptimeCheck configuration.
+type UptimeCheck struct {
+	ID      string   `json:"id"`
+	Name    string   `json:"name"`
+	Type    string   `json:"type"`
+	Target  string   `json:"target"`
+	Regions []string `json:"regions"`
+	Enabled bool     `json:"enabled"`
+}
+
+// UptimeAlert represents a DigitalOcean Uptime Alert configuration.
+type UptimeAlert struct {
+	ID            string         `json:"id"`
+	Name          string         `json:"name"`
+	Type          string         `json:"type"`
+	Threshold     int            `json:"threshold"`
+	Comparison    string         `json:"comparison"`
+	Notifications *Notifications `json:"notifications"`
+	Period        string         `json:"period"`
+}
+
+// Notifications represents a DigitalOcean Notifications configuration.
+type Notifications struct {
+	Email []string       `json:"email"`
+	Slack []SlackDetails `json:"slack"`
+}
+
+// UptimeCheckState represents a DigitalOcean Uptime Check's state configuration.
+type UptimeCheckState struct {
+	Regions        UptimeRegions        `json:"regions"`
+	PreviousOutage UptimePreviousOutage `json:"previous_outage"`
+}
+
+type UptimeRegions struct {
+	USEast UptimeRegion `json:"us_east"`
+	USWest UptimeRegion `json:"us_west"`
+}
```

Comment:
```suggestion
type UptimeCheckState struct {
	Regions        map[string]UptimeRegion `json:"regions"`
	PreviousOutage UptimePreviousOutage    `json:"previous_outage"`
}
```

Despite only `us_east` and `us_west` being mentioned in the docs, it looks like there are additional regions. Checking an alert on my account also shows `eu_west` and  `se_asia`. It might be more maintainable to use a `map[string]UptimeRegion` here.

Reply:
oh gosh you're right- thanks